src/processor/operator/recursive_extend/path_scanner.cpp

```diff
@@ -0,0 +1,77 @@
+#include "processor/operator/recursive_extend/path_scanner.h"
+
+namespace kuzu {
+namespace processor {
+
+bool FixedSizePathScanner::getNext() {
```

Comment:
I have a few comments:
1) You seem to have different DFSScanner for different depths. This is not really necessary and would traverse the smaller depth paths multiple times. I am guessing you are doing this because you want to output all 3-length paths, and then all 4-length paths, etc.. But why do you have to do this (I vaguely recall you saying Neo4j does this but I don't think we have to do it). 

2) You are using a recursive implementation of DFS, which is destined to be very inefficient. You should try to have stack-based iterative implementation. Something based on the core iterative DFS algorithm, if you really have to do DFS. 

3) I also see that in BaseRecursiveJoin::scanOutput() function you are outputting a  vector of paths. But you are not doing anything in the DFS to output a vector of paths. You still do one step of DFS at a time. You can modify your DFS, so that for a prefix of nodes, you can do the "DFS traversal" one adjacency list at a time and output.

These are common wisdom implementation tricks for fast dfs. You might want to benchmark to measure the effects.